hail/src/main/scala/is/hail/expr/ir/NormalizeNames.scala

```diff
@@ -241,10 +241,18 @@ class NormalizeNames(normFunction: Int => String, allowFreeVariables: Boolean =
           newDynamicID <- normalize(dynamicID)
         } yield CollectDistributedArray(newCtxs, newGlobals, newC, newG, newBody, newDynamicID, staticID, tsd)
       case RelationalLet(name, value, body) =>
+        val newName = gen()
         for {
-          newValue <- normalize(value, BindingEnv.empty)
-          newBody <- normalize(body)
-        } yield RelationalLet(name, newValue, newBody)
+          newValue <- normalize(value, env)
+          newBody <- normalize(body, env.bindRelational(name, newName))
+        } yield RelationalLet(newName, newValue, newBody)
```

Comment:
This should match the binding structure defined in `Binds.scala`, which would be
```
          newValue <- normalize(value, env.onlyRelational)
          newBody <- normalize(body, env.noAgg.noScan.bindRelational(name, newName))
```

Reply:
Thanks